src/main/java/uk/gov/justice/digital/delius/service/RequirementService.java

```diff
@@ -29,18 +37,55 @@ public ConvictionRequirements getRequirementsByConvictionId(String crn, Long con
         var offender = offenderRepository.findByCrn(crn)
                 .orElseThrow(() -> new NotFoundException(String.format("Offender with CRN '%s' not found", crn)));
 
-        var requirements = eventRepository.findByOffenderId(offender.getOffenderId())
-                .stream()
-                .filter(event -> convictionId.equals(event.getEventId()))
-                .map(Event::getDisposal)
-                .flatMap(this::getRequirementStreamFromDisposal)
+        var requirements = getDisposalStream(convictionId, offender)
+                .flatMap(this::getRequirementStream)
                 .map(RequirementTransformer::requirementOf)
                 .collect(Collectors.toList());
 
         return new ConvictionRequirements(requirements);
     }
 
-    private Stream<Requirement> getRequirementStreamFromDisposal(Disposal disposal) {
-        return disposal != null && disposal.getRequirements() != null ? disposal.getRequirements().stream() : Stream.empty();
+    public PssRequirements getPssRequirementsByConvictionId(String crn, Long convictionId) {
+        var offender = offenderRepository.findByCrn(crn)
+                .orElseThrow(() -> new NotFoundException(String.format("Offender with CRN '%s' not found", crn)));
+
+        var requirements = getDisposalStream(convictionId, offender)
+                .map(Disposal::getCustody)
```

Comment:
very minor, but a bit clunky using a list of Disposals and Custody when there can only be one - but I suppose the result is correct - so perhaps not worry about it

Reply:
No this is a good call, it did occur to me that some of this stream processing seems out of place given we're only expecting one element.  I think I started with a list of events and then ran with it but now that you mention it, it probably would make more sense to resolve it to a single event and then only start streaming again when you get the pssRequirements. Will amend 👍